domain/application/state/application.go

```diff
@@ -227,15 +227,23 @@ func (st *State) CreateApplication(
 			}
 		}
 
+		modelType, err := st.GetModelType(ctx)
```

Comment:
Shouldn't this get pushed down further to where it is needed?

Reply:
Yes but then it would be looked up again and again in a loop.
The insert unit code needs to be refactored which is a known issue. That can happen at some point after this lands. We can't do it all in one PR.